X10D/src/Core/SpanExtensions.cs

```diff
@@ -317,29 +346,33 @@ public static unsafe int PackInt32(this ReadOnlySpan<bool> source)
+                        var load = Sse2.LoadVector128((byte*)pSource);
+                        var correct = IntrinsicUtility.CorrectBoolean(load).AsUInt64();
 
-                        var magic = Vector128.Create(IntegerPackingMagic);
+                        var multiply = IntrinsicUtility.Multiply(IntegerPackingMagicV128, correct);
+                        var shift1 = Sse2.ShiftRightLogical(multiply, 56);
+                        shift1 = Sse2.ShiftLeftLogical(shift1, Vector128.Create(0UL, 8UL));
 
-                        var calc1 = Sse2.ShiftRightLogical(IntrinsicUtility.Multiply(magic, vector1), 56);
-                        var calc2 = Sse2.ShiftRightLogical(IntrinsicUtility.Multiply(magic, vector2), 56);
+                        load = Sse2.LoadVector128((byte*)(pSource + 16));
+                        correct = IntrinsicUtility.CorrectBoolean(load).AsUInt64();
 
-                        var shift1 = Sse2.ShiftLeftLogical(calc1, Vector128.Create(0UL, 8UL));
-                        var shift2 = Sse2.ShiftLeftLogical(calc2, Vector128.Create(16UL, 24UL));
+                        multiply = IntrinsicUtility.Multiply(IntegerPackingMagicV128, correct);
+                        var shift2 = Sse2.ShiftRightLogical(multiply, 56);
+                        shift2 = Sse2.ShiftLeftLogical(shift2, Vector128.Create(16UL, 24UL));
 
-                        return (int)(shift1.GetElement(0) | shift1.GetElement(1) | shift2.GetElement(0) | shift2.GetElement(1));
+                        var or1 = Sse2.Or(shift1, shift2);
+                        var or2 = Sse2.Or(or1, IntrinsicUtility.ReverseElements(or1));
+
+                        return (int)or2.GetElement(0);
                     }
-                    else if (AdvSimd.IsSupported)
+                    if (AdvSimd.IsSupported)
                     {
                         // Hasn't been tested since March 6th 2023 (Reason: Unavailable hardware).
-                        var vector1 = AdvSimd.LoadVector128((byte*)pSource).CorrectBoolean().AsUInt64();
-                        var vector2 = AdvSimd.LoadVector128((byte*)(pSource + 16)).CorrectBoolean().AsUInt64();
-
-                        var magic = Vector128.Create(IntegerPackingMagic);
+                        var vector1 = IntrinsicUtility.CorrectBoolean(AdvSimd.LoadVector128((byte*)pSource)).AsUInt64();
+                        var vector2 = IntrinsicUtility.CorrectBoolean(AdvSimd.LoadVector128((byte*)(pSource + 16))).AsUInt64();
 
-                        var calc1 = AdvSimd.ShiftRightLogical(IntrinsicUtility.Multiply(magic, vector1), 56);
-                        var calc2 = AdvSimd.ShiftRightLogical(IntrinsicUtility.Multiply(magic, vector2), 56);
+                        var calc1 = AdvSimd.ShiftRightLogical(IntrinsicUtility.Multiply(IntegerPackingMagicV128, vector1), 56);
+                        var calc2 = AdvSimd.ShiftRightLogical(IntrinsicUtility.Multiply(IntegerPackingMagicV128, vector2), 56);
```

Comment:
A lot of use of `var` in this file. I have no idea what types these variables are, it's not apparent